### PR TITLE
feat(Agent Configuration):Add an Endpoint to lists available pollers

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description

This PR intends to add an endpoint to list available pollers for Agent Configuration

### Behaviour
Agent Pollers are pollers not already linked to an Agent Configuration.
An admin could see any available pollers
An non admin could only see pollers in his ACLs groups

### Tests
```
   PASS  Tests\php\Core\AgentConfiguration\Application\UseCase\FindPollers\FindPollersTest
  ✓ it should present an ErrorResponse when an exception is thrown
  ✓ it should retrieve poller with no ACLs calculation for an admin user
  ✓ it should retrieve poller with ACLs calculation for a non admin user
  ✓ it should present a FindPollersResponse when no errors occurred
```

### API Return
```json
{
  "result": [
    {
      "id": 1,
      "name": "Central"
    },
    {
      "id": 2,
      "name": "aaaa"
    }
  ],
  "meta": {
    "page": 1,
    "limit": 10,
    "search": {},
    "sort_by": {},
    "total": 2
  }
}
```
**Fixes** # MON-150044
## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [ ] 24.04.x
- [x] master

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by dmyios — APPROVED** [review]

File: centreon/src/Core/AgentConfiguration/Application/UseCase/FindPollers/FindPollers.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                : $this->readRepository->findAvailablePollersByRequestParametersAndAccessGroups(
+                    $this->readAccessGroupRepository->findByContact($this->user),
+                    $this->requestParameters
+                );
+
+            return new FindPollersResponse($pollers);
+        } catch (\Throwable $ex) {
+            $this->error($ex->getMessage(), ['trace' => $ex->getTraceAsString()]);
+
+            return new ErrorResponse(AgentConfigurationException::findPollers());
```
Comment (adr-mo) [inline centreon/src/Core/AgentConfiguration/Application/UseCase/FindPollers/FindPollers.php:67]: I find it kind of disturbing here the name of the static method called for the exception 😮 
I know for sure that we used to explicit what went wrong for instance errorOnSearch and so on...
_(That's just a comment)_

File: centreon/src/Core/AgentConfiguration/Infrastructure/Voters/AgentConfigurationVoter.php [hunk 2]
```diff
@@ -41,6 +41,7 @@ final class AgentConfigurationVoter extends Voter
     use LoggerTrait;
     public const READ_AC = 'read_agent_configuration';
     public const READ_AC_POLLERS = 'read_agent_configuration_pollers';
+    public const READ_AVAILABLE_POLLERS = 'read_agent_configuration_available_pollers';
```
Comment (adr-mo) [inline centreon/src/Core/AgentConfiguration/Infrastructure/Voters/AgentConfigurationVoter.php:44]: Same logic could have been kept here no ?
```suggestion
    public const REAC_AC_AVAILABLE_POLLERS = 'read_agent_configuration_available_pollers';
```

Comment (jeremyjaouen) [reply centreon/src/Core/AgentConfiguration/Infrastructure/Voters/AgentConfigurationVoter.php:44]: indeed

File: centreon/src/Core/AgentConfiguration/Infrastructure/Voters/AgentConfigurationVoter.php [hunk 3]
```diff
@@ -79,7 +80,7 @@ protected function voteOnAttribute(string $attribute, $subject, TokenInterface $
         }
 
         return match ($attribute) {
-            self::READ_AC => $this->checkTopologyRole($user),
+            self::READ_AC, self::READ_AVAILABLE_POLLERS => $this->checkTopologyRole($user),
```
Comment (a-launois) [inline centreon/src/Core/AgentConfiguration/Infrastructure/Voters/AgentConfigurationVoter.php:83]: Why not reuse the READ_AC ?

**Review by a-launois — APPROVED** [review]

**Review by callapa — COMMENTED** [review]
If the need is to display the list of pollers, this endpoint should be placed in the Core/MonitoringServer folder.

File: centreon/src/Core/AgentConfiguration/Application/Exception/AgentConfigurationException.php [hunk 4]
```diff
@@ -149,4 +149,9 @@ public static function onlyOnePoller(int $pollerId, int $acId): self
             self::CODE_CONFLICT
         );
     }
+
+    public static function findPollers(): self
+    {
+        return new self(_('Error while searching for available pollers'));
```
Comment (a-launois) [inline centreon/src/Core/AgentConfiguration/Application/Exception/AgentConfigurationException.php:155]: Missing translation

**Review by wtermellil — APPROVED** [review]

**Pull request closed without merging**